client/src/reducers/search-filters-reducer.js

```diff
@@ -0,0 +1,269 @@
+import _ from 'lodash';
+
+import { LOCATION_CHANGE } from 'react-router-redux'
+import Reducer from 'reducers/reducer';
+import store from 'app/store';
+
+import DateTransformer from 'lib-core/date-transformer';
+
+
+const TICKET_STATUSES = {
+    ANY: undefined,
+    OPENED: 0,
+    CLOSED: 1
+};
+
+const CLOSED_DROPDOWN_INDEXES = {
+    ANY: 0,
+    OPENED: 1,
+    CLOSED: 2
+}
+
+const TICKET_PRIORITIES = {
+    ANY: undefined,
+    LOW: [0],
+    MEDIUM: [1],
+    HIGH: [2]
+};
+
+const PRIORITIES_DROPDOWN_INDEXES = {
+    ANY: 0,
+    LOW: 1,
+    MEDIUM: 2,
+    HIGH: 3
+};
+
+const DEFAULT_FILTERS = {
+    query: undefined,
+    closed: undefined,
+    priority: undefined,
+    departments: "[]",
+    owners: "[]",
+    tags: "[]",
+    dateRange: getDefaultDateRangeForFilters(),
+    orderBy: undefined,
+};
+
+function getDefaultDateRangeForFilters() {
+    return JSON.stringify(DateTransformer.formDateRangeToFilters([20170101, DateTransformer.getDateToday()]));
+}
+
+const DEFAULT_START_DATE = 20170101;
+
+class searchFiltersReducer extends Reducer {
+    
+    getInitialState() {
+        return {
+            showFilters: false,
+            listData: this.getList(),
+            form: {
+                query: "",
+                closed: 0,
+                priority: 0,
+                departments: [],
+                owners: [],
+                tags: [],
+                dateRange: {valid: true, startDate: DEFAULT_START_DATE, endDate: DateTransformer.getDateToday()},    
+            }
+        };
+    }
+
+    getTypeHandlers() {
+        return {
+            'SEARCH_FILTERS_ON_SUBMIT_FORM': this.onSubmitForm.bind(this),
+            'SEARCH_FILTERS_CHANGE_FORM': this.onFormChange,
+            'SEARCH_FILTERS_CHANGE_FILTERS': this.onFiltersChange.bind(this),
+            'SEARCH_FILTERS_SET_DEFAULT_FORM_VALUES': this.onSetDefaultFormValues.bind(this),
+            'SEARCH_FILTERS_CHANGE_SHOW_FILTERS': this.onChangeShowFilters.bind(this),
```

Comment:
please put declaration of this functions directly after` getTypeHandlers()`

Reply:
okey